src/gui_common/tooltip/ToolTipManager.cs

```diff
@@ -296,33 +321,48 @@ private void FetchToolTips()
 
             foreach (ICustomToolTip tooltip in group.GetChildren())
             {
-                tooltip.ToolTipVisible = false;
+                tooltip.ToolTipNode.Visible = false;
                 collectedTooltips.Add(tooltip);
             }
 
             tooltips.Add(group, collectedTooltips);
         }
     }
 
-    private void UpdateToolTipVisibility()
+    private void FinalizeToolTipVisibility(ICustomToolTip tooltip, bool visible)
     {
-        // TODO: Fix the current tooltip changing while still fading out
-        // when quickly mousing over multiple closely positioned elements
-        // (Happens when a single tooltip is registered to multiple Controls)
+        if (tooltip.ToolTipNode.Visible == visible)
+            return;
 
-        // Make sure to hide any other tooltips that are still visible
-        HideAllToolTips();
+        // TODO: Fix fade type transition when tooltip display delay is less than the tooltip fade speed,
+        // some kind of flickering happens
 
-        if (Display)
+        switch (tooltip.TransitionType)
         {
-            // Set timer
-            displayTimer = MainToolTip.DisplayDelay;
+            case ToolTipTransitioning.Immediate:
+                tooltip.ToolTipNode.Visible = visible;
+                break;
+            case ToolTipTransitioning.Fade when visible:
+                GUICommon.Instance.ModulateFadeIn(tooltip.ToolTipNode, Constants.TOOLTIP_FADE_SPEED);
+                break;
+
+            // ReSharper disable once ConditionIsAlwaysTrueOrFalse
```

Comment:
Why's this here?
I'm about to open a PR where the jetbrains version is updated to 2021.2, might that help?

Reply:
Also first time when I see this `when` syntax. So if the check doesn't pass otherwise how about combining these to a one case with an if inside it?

Reply:
Just knew about it as well but CI of course just doesn't like it. The if block inside one case is the original form I intended, will just revert to that.

Reply:
Pretty sure the jb complaint was significant. removing the `when !visible` probably should have fixed it. This is because the case above already handles the fade case when visible, so here the when condition is unnecessary.